src/rendergraph/common/rendergraph/material/rgbmaterial.cpp

```diff
@@ -0,0 +1,39 @@
+#include "rgbmaterial.h"
+
+#include <QVector2D>
+
+#include "rendergraph/materialshader.h"
+#include "rendergraph/materialtype.h"
+#include "rendergraph/uniformset.h"
+
+using namespace rendergraph;
+
+RGBMaterial::RGBMaterial()
+        : Material(uniforms()) {
+}
+
+/* static */ const AttributeSet& RGBMaterial::attributes() {
+    static AttributeSet set = makeAttributeSet<QVector2D, QVector3D>({"position", "color"});
+    return set;
+}
+
+/* static */ const UniformSet& RGBMaterial::uniforms() {
+    static UniformSet set = makeUniformSet<QMatrix4x4>({"ubuf.matrix"});
+    return set;
+}
+
+MaterialType* RGBMaterial::type() const {
+    static MaterialType type;
+    return &type;
+}
+
+int RGBMaterial::compare(const Material* other) const {
+    Q_ASSERT(other && type() == other->type());
+    const auto* otherCasted = static_cast<const RGBMaterial*>(other);
```

Comment:
You are absolutely right. It does work because in the end we don't do anything with the pointer other than comparing its value, but conceptually this is wrong. Well spotted.